pkg/ruler/rule_concurrency.go

```diff
@@ -0,0 +1,213 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package ruler
+
+import (
+	"context"
+	"sync"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+	"github.com/grafana/dskit/tenant"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/prometheus/rules"
+	"golang.org/x/sync/semaphore"
+)
+
+type MultiTenantRuleConcurrencyController interface {
+	rules.RuleConcurrencyController
+}
+
+const IntervalToEvaluationThreshold = 50.00
```

Comment:
Can you make this configurable (**not** per tenant), please? I want to exercise this code in dev, but with a threshold of 50% it will be never exercised.

Reply:
I think I got it as part of https://github.com/grafana/mimir/pull/8146/commits/f419dc09cc88b80e4db9b583870e88dffdf32266 let me know what you think.